Kernel/Bus/USB/UHCI/UHCIDescriptorTypes.h

```diff
@@ -69,21 +69,29 @@ struct alignas(16) TransferDescriptor final {
         ShortPacketDetect = (1 << 29),
     };
 
+    struct TransferDescriptorBookkeeping {
+        u32 paddr;                               // Physical 4-byte address where this TransferDescriptor is located
+        TransferDescriptor* next_td { nullptr }; // Pointer to first TD
+        TransferDescriptor* prev_td { nullptr }; // Pointer to the previous TD
+        bool in_use;                             // Has this TD been allocated (and therefore in use)?
+    };
+
     TransferDescriptor() = delete;
     TransferDescriptor(u32 paddr)
-        : m_paddr(paddr)
     {
+        m_bookkeeping = new TransferDescriptorBookkeeping;
```

Comment:
This PR has been around a while and is probably fine to merge but, these two calls to `::new` stand out to me now with two questions:

1) when are they freed? 
2) why can it not fail?

Are these descriptors all allocated during init? Or are we calling unfallible new wily nilly as someone hot plugs their USB devices?

Reply:
1. These are never freed. The descriptors are kept around in a pool and allocated via the descriptor pool allocator function.
2. I suppose it can fail if the system OOMs on init. If that's the case, this is the last of your concern.

And yep, they're all allocated during init: 

https://github.com/SerenityOS/serenity/blob/master/Kernel/Bus/USB/UHCI/UHCIDescriptorPool.h#L76-L80
https://github.com/SerenityOS/serenity/blob/master/Kernel/Bus/USB/UHCI/UHCIDescriptorPool.h#L76-L80
https://github.com/SerenityOS/serenity/blob/master/Kernel/Bus/USB/UHCI/UHCIController.cpp#L83
https://github.com/SerenityOS/serenity/blob/30b58cd06ccb1855d365d362719cac4f8867c72b/Kernel/Bus/USB/USBManagement.cpp#L37

Reply:
aight let's get this in then so we can move fwd on the other USB PRs